MiniApp/Classes/Storage/MiniAppLocalServerPreferences.swift

```diff
@@ -0,0 +1,15 @@
+class MiniAppLocalServerPreferences {
+    private let defaults: UserDefaults?
+
+    init() {
+        self.defaults = UserDefaults(suiteName: "com.rakuten.tech.mobile.miniapp.server")
+    }
+
+    func savePortNumberForMiniApp(appId: String, portNumber: Int) {
+        defaults?.setValue(portNumber, forKey: appId)
+    }
+
+    func getPortNumberForMiniApp(key: String) -> Int {
```

Comment:
One thing I haven't been able to understand: How is the port number generated for a Mini App?

We need to make sure that we don't conflict with the default ports on iOS: https://support.apple.com/en-us/HT202944 Also, the max port number is `65535`. Is there anything in place to prevent conflicts or too high of a port number?

Reply:
If we send the port # as `0` and when mini-app is launched for the first time, the local server will start with random available port number (less than `65535`) and the same is stored in defaults. Next time when the mini-app is launched we will use the same port #

Reply:
Ah, okay nice so it's just generated by the Telegraph library?

But isn't it still possible that it will generate the same port number twice? For example, if you start a Mini App once, then close the Host App, and then start a different Mini App sometime in the future, it could generate (by chance) the same random number for the other Mini App. 

The library doesn't have a cache of which port numbers it has used, right? Should there be some checking added to make sure that the generated port isn't already being used by another Mini App?

Reply:
By the way, do we really need one port by miniApp ? That virtually means one server by miniApp, right ?

Reply:
Each one has a different port so that they can have data separation for localStorage, etc. Is there another way to do that without using different ports?

Also, I think that the server is stopped when the Mini App is closed, right?

Reply:
Oh yes server is stopped during the removeFromSuperView, it was not the case during our very first tests so that's why I wrote this, sorry